src/common/utility.ts

```diff
@@ -92,6 +95,78 @@ export class Utility {
         return mysql.createConnection(newConnectionOptions);
     }
 
+    public static async editConnection(connectionNode: ConnectionNode, context: vscode.ExtensionContext, mysqlTreeDataProvider: MySQLTreeDataProvider) {
+        if (connectionNode) {
+            connectionNode.editConnection(context, mysqlTreeDataProvider);
+        } else {
+            // No connection has been selected, let the user pick one
+            const connections = context.globalState.get<{ [key: string]: IConnection }>(Constants.GlobalStateMySQLConectionsKey);

```

Comment:
Pick connection logic is similar in deleteConnection and editConnection. Please merge them